assets/js/components/SapSystemsOverview/SapSystemsOverview.test.jsx

```diff
@@ -2,28 +2,202 @@ import React from 'react';
 import { screen, waitFor } from '@testing-library/react';
 import 'intersection-observer';
 import '@testing-library/jest-dom';
-import { sapSystemFactory } from '@lib/test-utils/factories';
+import {
+  clusterFactory,
+  hostFactory,
+  sapSystemFactory,
+} from '@lib/test-utils/factories';
 import { renderWithRouter, withState } from '@lib/test-utils';
 import { filterTable, clearFilter } from '@components/Table/Table.test';
 
 import SapSystemsOverview from './SapSystemsOverview';
 
+const cleanInitialState = {
+  hostsList: {
+    hosts: [],
+  },
+  clustersList: {
+    clusters: [],
+  },
+  sapSystemsList: {
+    sapSystems: [],
+    applicationInstances: [],
+    databaseInstances: [],
+  },
+};
+
 describe('SapSystemsOverviews component', () => {
-  describe('filtering', () => {
-    const cleanInitialState = {
-      hostsList: {
-        hosts: [],
-      },
-      clustersList: {
-        clusters: [],
-      },
-      sapSystemsList: {
-        sapSystems: [],
-        applicationInstances: [],
-        databaseInstances: [],
-      },
-    };
+  describe('overview content', () => {
+    it('should display the correct number of SAP systems', () => {
+      const sapSystemCount = 3;
+      const expectedRowCount = sapSystemCount * 2;
+      const state = {
+        ...cleanInitialState,
+        sapSystemsList: {
+          sapSystems: sapSystemFactory.buildList(sapSystemCount),
+          applicationInstances: [],
+          databaseInstances: [],
+        },
+      };
+
+      const [StatefulSapSystemList] = withState(<SapSystemsOverview />, state);
+
+      renderWithRouter(StatefulSapSystemList);
+
+      expect(
+        screen.getByRole('table').querySelectorAll('tbody > tr')
+      ).toHaveLength(expectedRowCount);
+    });
 
+    it('should display the correct content for a SAP system main row', () => {
+      const sapSystem = sapSystemFactory.build({ ensa_version: 'ensa1' });
+      const {
+        id: sapSystemID,
+        sid,
+        tenant,
+        db_host: dbAddress,
+        application_instances: applicationInstances,
+        database_instances: databaseInstances,
+      } = sapSystem;
+
+      const state = {
+        ...cleanInitialState,
+        sapSystemsList: {
+          sapSystems: [sapSystem],
+          applicationInstances,
+          databaseInstances,
+        },
+      };
+
+      const [StatefulSapSystemList] = withState(<SapSystemsOverview />, state);
+
+      renderWithRouter(StatefulSapSystemList);
+
+      const rows = screen.getByRole('table').querySelectorAll('tbody > tr');
```

Comment:
This approach is fine but you could also just do a query for 'ENSA1', 'ENSA2' and count the number of nodes that appear

Reply:
Yes, as well.
I wanted to have a more detailed test, where I check the cell position within the table together with the content